### PR TITLE
Drop pull-kubernetes-e2e-gce-providerless CI job for master ONLY

### PR DESCRIPTION
**Pull request by dims**
We don't have `providerless` tag any more since https://github.com/kubernetes/kubernetes/pull/124519 landed.

### CODE REVIEW

**Review by upodroid — APPROVED** [review]
LGTM

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.30.yaml [hunk 1]
```diff
@@ -814,13 +814,19 @@ presubmits:
       org: kubernetes
       path_alias: k8s.io/release
       repo: release
+    annotations:
+      fork-per-release: "true"
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.30.yaml:818]: the fork annotations needed to be dropped

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.30.yaml:818]: https://github.com/kubernetes/test-infra/pull/33277

**Pull request merged**